src/pf/existing/detect_base_dir.cpp

```diff
@@ -0,0 +1,53 @@
+#include "./detect_base_dir.hpp"
+
+#include <algorithm>
+#include <cstdlib>  // std::abort
+#include <fstream>
+#include <string>
+#include <utility>
+
+#include <boost/range/iterator_range.hpp>
+
+namespace fs = pf::fs;
+
+namespace {
+std::optional<fs::path> parse_cmakecache_homedir(fs::path const& cmakecache) {
+    std::string line;
+
+    std::ifstream file{cmakecache};
```

Comment:
I'd go with `pf::open()` as it will do error checking, otherwise this will just silently fail.